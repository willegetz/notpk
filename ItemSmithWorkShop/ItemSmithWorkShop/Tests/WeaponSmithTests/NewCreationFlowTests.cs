using System;
using System.Text;
using System.Collections.Generic;
using System.Linq;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using ApprovalTests;
using ItemSmithWorkShop.WeaponSmith;

namespace ItemSmithWorkShop.Tests.WeaponSmithTests
{
	[TestClass]
	public class NewCreationFlowTests
	{
		[TestMethod]
		public void TestCreateWeaponClass()
		{
			var weaponData = TempWeaponDictionary.GetWeaponData("Dagger");

			var dagger = new CreateWeapon(weaponData);

			Approvals.Approve(dagger.DisplayWeapon1());
		}

		[TestMethod]
		public void TestCreateDoubleWeaponWithNewClass()
		{
			var weaponData1 = TempWeaponDictionary.GetWeaponData("Dwarven Urgrosh A");
			var weaponData2 = TempWeaponDictionary.GetWeaponData("Dwarven Urgrosh B");

			var weapon = new CreateWeapon(weaponData1, weaponData2);

			Approvals.Approve(weapon.DisplayWeapon());
		}
	}
}
