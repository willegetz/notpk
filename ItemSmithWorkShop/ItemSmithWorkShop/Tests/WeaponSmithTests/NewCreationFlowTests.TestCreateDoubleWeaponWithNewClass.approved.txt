Dwarven Urgrosh (Medium)
Two-Handed Melee Weapon
Exotic Weapon Proficiency
Weight: 12 pound(s)
Total Weapon Cost: 50 gold pieces

Dwarven Urgrosh Axe Head
Damage: 1d8 [20 x2] Slashing
Hardness: 5
Hit Points: 10
Weapon Text:
	A dwarven urgrosh is a double weapon.	This is the axe head.

Dwarven Urgrosh Spear Head
Damage: 1d6 [20 x2] Piercing
Hardness: 5
Hit Points: 10
Weapon Text:
	A dwarven urgrosh is a double weapon.	This is the spear head.